### PR TITLE
Support to run root permission job

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit supports to run a job that requires root permission,
such as setting up the toolchain envrionment.

ONE-vscode-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

Draft pr: #599

### CODE REVIEW

**Review by hyunsik-yoon — APPROVED** [review]

File: src/Project/ToolRunner.ts [hunk 1]
```diff
@@ -83,10 +83,20 @@ export class ToolRunner {
...
+      if (root) {
+        // NOTE
+        // To run the root command job, it must requires a password in `process.env.userp`
+        // environment.
+        // TODO(jyoung): Need password encryption
+        tool = `echo ${process.env.userp} | sudo -S ` + tool;
+      }
+      let cmd = cp.spawn(tool, toolargs, {cwd: path, shell: true});
+      if (root) {
+        process.env.userp = '';
```
Comment (llFreetimell) [inline src/Project/ToolRunner.ts:98]: Can I ask you why this is reset?
And isn't it possible to integrate two if-statement? :D

File: src/Project/ToolRunner.ts [hunk 2]
```diff
@@ -83,10 +83,20 @@ export class ToolRunner {
...
+      }
```
Comment (jyoungyun) [inline src/Project/ToolRunner.ts:99]:
```suggestion
      if (root) {
        // NOTE
        // To run the root command job, it must requires a password in `process.env.userp`
        // environment.
        // TODO(jyoung): Need password encryption
        tool = `echo ${process.env.userp} | sudo -S ` + tool;
        let cmd = cp.spawn(tool, toolargs, {cwd: path, shell: true});
        process.env.userp = '';
      } else {
        let cmd = cp.spawn(tool, toolargs, {cwd: path});
      }
```
> And isn't it possible to integrate two if-statement? :D

Do you mean this code?

Comment (llFreetimell) [reply src/Project/ToolRunner.ts:99]: Sort of :D
Of course, it is not mandatory!

File: src/Project/ToolRunner.ts (hunk 1 above)
Comment (jyoungyun) [reply src/Project/ToolRunner.ts:98]: Becasue this data is the password of the user machine. So I think it should be removed for safety. Actually I don't want to store this information, but there is no way to pass this value, so I stored it temporarily.

**Review by llFreetimell — APPROVED** [review]
LGTM

**Pull request merged**